ddtrace/contrib/asyncio/patch.py

```diff
@@ -1,39 +1,27 @@
-# project
-import ddtrace
-from ddtrace.util import unwrap
-from ddtrace.provider import DefaultContextProvider
-
-# 3p
-import wrapt
 import asyncio
 
-from .helpers import _wrapped_create_task
-from . import context_provider
+from wrapt import wrap_function_wrapper as _w
 
-_orig_create_task = asyncio.BaseEventLoop.create_task
+from .helpers import _wrapped_create_task
+from ...util import unwrap as _u
 
 
-def patch(tracer=ddtrace.tracer):
-    """
-    Patches `BaseEventLoop.create_task` to enable spawned tasks to parent to
-    the base task context.  Will also enable the asyncio task context.
+def patch():
+    """Patches current loop `create_task()` method to enable spawned tasks to
+    parent to the base task context.
     """
-    # TODO: figure what to do with helpers.ensure_future and
-    # helpers.run_in_executor (doesn't work for ProcessPoolExecutor)
     if getattr(asyncio, '_datadog_patch', False):
         return
     setattr(asyncio, '_datadog_patch', True)
 
-    tracer.configure(context_provider=context_provider)
-    wrapt.wrap_function_wrapper('asyncio', 'BaseEventLoop.create_task', _wrapped_create_task)
+    loop = asyncio.get_event_loop()
```

Comment:
Because not all the time the `BaseEventLoop` is used, we take the current event loop implementation and patch directly the real `create_task()` that is used.